### ISSUE TITLE
Error when trying to use in one of the Angular SPA 

### ISSUE THREAD

**Issue by umarkhan17** [jufab/opentelemetry-angular-interceptor#91]
Facing this error :
ARNING: Running a server with --disable-host-check is a security risk. See https://medium.com/webpack/webpack-dev-server-middleware-security-issues-1489d950874a for more information.   
WARNING: --deploy-url and/or --base-href contain unsupported values for ng serve. Default serve path of '/' used. Use --serve-path to override.
Compiling @jufab/opentelemetry-angular-interceptor : module as esm5
Warning: Invalid constructor parameter decorator in D:/cc/cc-new/RAT_cc-2.0/Micro Front End V2/navbar-app/node_modules/@jufab/opentelemetry-angular-interceptor/fesm2015/jufab-opentelemetry-angular-interceptor.js:
 () => [
    { type: undefined, decorators: [{ type: Inject, args: [OTLP_CONFIG,] }] },
    { type: undefined, decorators: [{ type: Inject, args: [OTELCOL_EXPORTER,] }] },
    { type: undefined, decorators: [{ type: Inject, args: [OTELCOL_PROPAGATOR,] }] },        
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OTLP_LOGGER,] 
}] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [CUSTOM_SPAN,] 
}] },
    { type: PlatformLocation }
]
Error: Error on worker #1: Error: getInternalNameOfClass() called on a non-ES5 class: expected OpenTelemetryHttpInterceptor to have an inner class declaration
    at Esm5ReflectionHost.getInternalNameOfClass (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\host\esm5_host.js:88:23)
    at DelegatingReflectionHost.getInternalNameOfClass (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\host\delegating_host.js:89:34)  
    at extractInjectableMetadata (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\src\ngtsc\annotations\src\injectable.js:125:69)
    at InjectableDecoratorHandler.analyze (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\src\ngtsc\annotations\src\injectable.js:66:24)        
    at NgccTraitCompiler.TraitCompiler.analyzeTrait (D:\cc\cc-new\RAT_cc-2.0\Micro Front End 
V2\navbar-app\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:345:40)
    at analyze (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:297:58)
    at _loop_1 (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:319:21)
    at NgccTraitCompiler.TraitCompiler.analyzeClass (D:\cc\cc-new\RAT_cc-2.0\Micro Front End 
V2\navbar-app\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:325:35)
    at NgccTraitCompiler.analyzeFile (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\analysis\ngcc_trait_compiler.js:47:26)
    at DecorationAnalyzer.analyzeProgram (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\analysis\decoration_analyzer.js:134:39)       
    at ClusterMaster.onWorkerMessage (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:194:27)
    at D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:54:95
    at ClusterMaster.<anonymous> (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:292:57)
    at step (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\tslib\tslib.js:136:27)
    at Object.next (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\tslib\tslib.js:117:57)
    at D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\tslib\tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\tslib\tslib.js:106:16)
    at EventEmitter.<anonymous> (D:\cc\cc-new\RAT_cc-2.0\Micro Front End V2\navbar-app\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:286:32)
    at EventEmitter.emit (events.js:400:28)
An unhandled exception occurred: NGCC failed.
See "C:\Users\UmarK\AppData\Local\Temp\ng-tG6sIl\angular-errors.log" for further details.    
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file C:\Windows\system32\cmd.exe
npm ERR! errno ENOENT
npm ERR! navbar@0.0.0 serve:single-spa: `ng serve --disable-host-check --port 4300 --deploy-url http://localhost:4300/ --live-reload false`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the navbar@0.0.0 serve:single-spa script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! navbar@0.0.0 start: `npm run serve:single-spa`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the navbar@0.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

Requesting for your help.

**Comment by jufab** [comment]
Hi @umarkhan17,

Can you be more explicit?
Which version (lib and angular)?

Thx

**Comment by umarkhan17** [comment]
Latest version of library.

**Comment by umarkhan17** [comment]
Angular 9 & latest library version

**Comment by umarkhan17** [comment]
Angular 9 & latest library version.
Could you please tell which angular version & library version this works best?

**Comment by jufab** [comment]
If you can, try with Angular 10. This lib is compiled with angular 10. 

For lib version, if you use latest release 1.0.0, add opentelemetry versions :

```
    "@opentelemetry/api": "1.0.3",
    "@opentelemetry/context-zone-peer-dep": "1.0.0",
    "@opentelemetry/exporter-otlp-http": "0.26.0",
    "@opentelemetry/exporter-zipkin": "1.0.0",
    "@opentelemetry/instrumentation": "0.26.0",
    "@opentelemetry/instrumentation-document-load": "0.26.0",
    "@opentelemetry/instrumentation-fetch": "0.26.0",
    "@opentelemetry/instrumentation-xml-http-request": "0.26.0",
    "@opentelemetry/propagator-aws-xray": "1.0.0",
    "@opentelemetry/propagator-b3": "1.0.0",
    "@opentelemetry/propagator-jaeger": "1.0.0",
    "@opentelemetry/sdk-trace-web": "1.0.0"
```

**Comment by umarkhan17** [comment]
Thanks. Will try. Also, considering that i am a beginner, i added configurations in Environment.ts, app.module.ts and added <otel-instrumentation> tag in app html. Will this automatically instrument my code or do i need to add any other configuration?

**Comment by umarkhan17** [comment]
I did the above mentioned steps, the error are gone but i am still unable to see traces.
Steps that i did :
upgraded to Angular 10,
installed all the required libraries,
Added configurations in Environment ts
Initialized in App.module.ts
Added <otel-instrumentation> in app.component.html
Used from instrumentation example.

Do i need to add any other configurations? 
P.S. : i am trying to implement on Angular SPA
